### PR TITLE
Fix Storage.filename to not track the filename when storage was mmap-ed with MAP_PRIVATE

### PR DESCRIPTION
**Pull request by mikaylagawarecki**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #128725

cc @albanD

### CODE REVIEW

File: torch/storage.py [hunk 1]
```diff
@@ -363,8 +363,10 @@ def is_hpu(self):
 
     @property
     def filename(self) -> _Optional[str]:
-        """Returns the file name associated with this storage if the storage was memory mapped from a file.
-           or ``None`` if the storage was not created by memory mapping a file."""
+        """Returns the file name associated with this storage.
+
+        The file name will be a string if the storage is on CPU and was either created
+        via :meth:`~torch.from_file()` with ``shared`` as ``True``. This attribute is ``None`` otherwise."""
```
Comment (albanD) [inline torch/storage.py:369]: You're missing the "or" for the "either"?

**Review by albanD — APPROVED** [review]
nit on doc but SGTM otherwise.

File: torch/storage.py (hunk 1 above)
Comment (mikaylagawarecki) [reply torch/storage.py:369]: oops there should be no either, thanks

**Pull request closed without merging**